src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -456,7 +456,40 @@ public static string GetPlatformApiLevel (string platformApiLevelName)
 
 		public static string GetLibraryImportDirectoryNameForAssembly (string assemblyIdentName)
 		{
-			return string.Concat (new MD2Managed ().ComputeHash (Encoding.UTF8.GetBytes (assemblyIdentName)).Take (5).Select (b => b.ToString ("X02")));
+			if (assemblyIdentityMap == null) {
+				throw new Exception ("Mapping was not loaded!");
+			}
+			if (assemblyIdentityMap.ContainsKey (assemblyIdentName)) {
+				return assemblyIdentityMap [assemblyIdentName];
+			}
+			string id = string.Format ("{0}", assemblyIdentityMap.Count + 1);
+			assemblyIdentityMap.Add (assemblyIdentName, id);
+			return id;
+		}
+
+		static Dictionary<string, string> assemblyIdentityMap;
```

Comment:
I'm willing to accept that overhead because I have no idea what the ramifications of "too much static data" is in the larger ecosystem. That *scares* me.